synapse/appservice/__init__.py

```diff
@@ -124,29 +125,23 @@ def _check_namespaces(self, namespaces):
                     raise ValueError(
                         "Expected bool for 'exclusive' in ns '%s'" % ns
                     )
-                if not isinstance(regex_obj.get("regex"), basestring):
+                regex = regex_obj.get("regex")
+                if isinstance(regex, basestring):
+                    regex_obj["regex"] = re.compile(regex)
```

Comment:
You're clobbering directly over a regex string with a regex object, but fail to say *why* you're doing this - which is to pre-compile it so you don't need to incur repeated compilation costs. A one-liner inline comment as you have there looks good to me.